### PR TITLE
10-sort-articles  is done

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
-import { useSearchParams } from "react-router-dom";
 import CreateTopicButton from "./CreateTopicButton";
 import ArticleList from "./article/ArticleList";
+import { useContext } from "react";
+import { NewsDataContext } from "../data/NewData";
 
 const Home = () => {
-    const [searchParams, setSearchParams] = useSearchParams();
+    const { searchParams } = useContext(NewsDataContext);
```
Comment (roisinoreilly) [inline src/components/Home.jsx:7]: nicely done

File: src/components/nav/Nav.jsx [hunk 2]
```diff
@@ -32,6 +36,27 @@ const Nav = () => {
...
+    }
+    function handleUrlOrder(order) {
+        const obj = { order: order };
+        if (params.topic) {
+            obj.topic = params.topic;
+        }
+        if (params.sort_by) {
+            obj.sort_by = params.sort_by;
+        }
+        setSearchParams(obj);
```
Comment (roisinoreilly) [inline src/components/nav/Nav.jsx:58]: really nice work

**Review by roisinoreilly — APPROVED** [review]
great job, Hassen! Your UI/UX is great and clearly well thought out, and functionality wise this is excellent

- Lovely UI to enable users to sort easily
- Good stuff including all the different sort options
- Excellent having the default sorting values in place
- Good sorting on both all articles and articles by a certain topic
- Some nice responsiveness in your designs - keep it up! 👍

- [x] I would suggest removing the sort dropdown (which I love by the way), or making it link to the all articles page, when the user is currently on a single article, because this is clickable on a single article but it does nothing

overall really excellent work! PR approved :)

File: src/components/nav/Nav.jsx (hunk 2 above)
Comment (Hassen-Ahmed) [reply src/components/nav/Nav.jsx:58]: Thanks so much @roisinoreilly!

**Pull request merged**